[Global Navigation] Clicked target is deleted after starting exercise

When clicking in the "play" button to run the exercise, the target that was marked in the GUI is removed (cleared). In previous versions the target stayed in the GUI after clicking "play".

It is ok to clear everything after clicking the "reset" button, but removing the target after clicking "play" is a bit painful for the user, because it requires setting a target while the exercise is running. Before, one could set the target and then run the exercise, which was much easier IMHO.

Hey, I just checked the remove-reset-target-global-nav branch and the problem still persists. When I click the "play" button, the target is removed.

If it helps, when selecting the target for the first time (before running the exerise), I get this error in the container terminal:

```console
[MainThread  ] [ERROR] (my_app_logger)  No client is connected.
```

Once the exercise is running, I no longer get that when clicking.

--- a/exercises/static/exercises/global_navigation_newmanager/python_template/ros2_humble/GUI.py
+++ b/exercises/static/exercises/global_navigation_newmanager/python_template/ros2_humble/GUI.py
@@ -44,7 +44,6 @@ class GUI(MeasuringThreadingGUI):
             self.mapXY = data
             x, y = self.mapXY
             self.worldXY = self.map.gridToWorld(x, y)
-            print(f"World : {self.worldXY}")
 
     # Prepares and sends a map to the websocket server
     def update_gui(self):

--- a/exercises/static/exercises/global_navigation_newmanager/react-components/GlobalNavigationSpecific.js
+++ b/exercises/static/exercises/global_navigation_newmanager/react-components/GlobalNavigationSpecific.js
@@ -119,10 +119,12 @@ function SpecificGlobalNavigation(props) {
         showMap = true
         setShowImage(true)
         // Resend Target
-        try {
-          window.RoboticsExerciseComponents.commsManager.send("gui", `pick${destinationWorld}`)
-        } catch (error) {
-        }  
+        if (destinationWorld) {
+          try {
+            window.RoboticsExerciseComponents.commsManager.send("gui", `pick${destinationWorld}`)
+          } catch (error) {
+          }  
+        }
       }
     };
     window.RoboticsExerciseComponents.commsManager.subscribe(

--- a/exercises/static/exercises/global_navigation_newmanager/react-components/GlobalNavigationSpecific.js
+++ b/exercises/static/exercises/global_navigation_newmanager/react-components/GlobalNavigationSpecific.js
@@ -114,7 +114,6 @@ function SpecificGlobalNavigation(props) {
         setShowImage(false)
         trail = []
         setPath("")
-        setDestination(undefined)
       } else {
         showMap = true
         setShowImage(true)

--- a/exercises/static/exercises/global_navigation_newmanager/react-components/GlobalNavigationSpecific.js
+++ b/exercises/static/exercises/global_navigation_newmanager/react-components/GlobalNavigationSpecific.js
@@ -10,6 +10,7 @@ function SpecificGlobalNavigation(props) {
   const [showImage, setShowImage] = React.useState(false);
   const [carPose, setCarPose] = React.useState(null)
   const [destination, setDestination] = React.useState(null)
+  const [destinationWorld, setDestinationWorld] = React.useState(null)
   const [path, setPath] = React.useState("")
   
   var trail = [];
@@ -117,6 +118,11 @@ function SpecificGlobalNavigation(props) {
       } else {
         showMap = true
         setShowImage(true)
+        // Resend Target
+        try {
+          window.RoboticsExerciseComponents.commsManager.send("gui", `pick${destinationWorld}`)
+        } catch (error) {
+        }  
       }
     };
     window.RoboticsExerciseComponents.commsManager.subscribe(
@@ -146,7 +152,7 @@ function SpecificGlobalNavigation(props) {
     let cursorYMap = cursorY / height;
 
     setDestination([(cursorY*100)/img.clientHeight, (cursorX*100)/(img.clientWidth*2)])
-
+    setDestinationWorld([cursorXMap, cursorYMap])
     return [cursorXMap, cursorYMap];
   }
 

--- a/exercises/static/exercises/global_navigation_newmanager/react-components/GlobalNavigationSpecific.js
+++ b/exercises/static/exercises/global_navigation_newmanager/react-components/GlobalNavigationSpecific.js
@@ -6,11 +6,12 @@ import CityMap from "../resources/images/cityLargenBin.png";
 
 import "./css/GUICanvas.css";
 
+var coords = undefined;
+
 function SpecificGlobalNavigation(props) {
   const [showImage, setShowImage] = React.useState(false);
   const [carPose, setCarPose] = React.useState(null)
   const [destination, setDestination] = React.useState(null)
-  const [destinationWorld, setDestinationWorld] = React.useState(null)
   const [path, setPath] = React.useState("")
   
   var trail = [];
@@ -119,9 +120,10 @@ function SpecificGlobalNavigation(props) {
         showMap = true
         setShowImage(true)
         // Resend Target
-        if (destinationWorld) {
+        console.log("Resend target:", coords)
+        if (coords) {
           try {
-            window.RoboticsExerciseComponents.commsManager.send("gui", `pick${destinationWorld}`)
+            window.RoboticsExerciseComponents.commsManager.send("gui", `pick${coords}`)
           } catch (error) {
           }  
         }
@@ -154,7 +156,6 @@ function SpecificGlobalNavigation(props) {
     let cursorYMap = cursorY / height;
 
     setDestination([(cursorY*100)/img.clientHeight, (cursorX*100)/(img.clientWidth*2)])
-    setDestinationWorld([cursorXMap, cursorYMap])
     return [cursorXMap, cursorYMap];
   }
 
@@ -163,6 +164,7 @@ function SpecificGlobalNavigation(props) {
       <img src={CityMap} alt="" className="exercise-canvas" id="exercise-img"
         onClick={ function pickLoc(event){
           var data = destinationPicker(event)
+          coords = data
           try {
             window.RoboticsExerciseComponents.commsManager.send("gui", `pick${data}`)
           } catch (error) {

